_rules/iframe-non-empty-title-attribute-cae760.md

```diff
@@ -0,0 +1,208 @@
+
+- [H64: Using the title attribute of the frame and iframe elements](https://www.w3.org/WAI/WCAG21/Techniques/html/H64)
+- [Understanding Success Criterion 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `iframe` element has a non-empty `title` [attribute value][].
+
+```html
+<iframe title="Grocery List" src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+#### Passed Example 2
+
+This `iframe` element with a [semantic role][] of `article` has a non-empty `title` [attribute value][].
+
+```html
+<iframe role="article" title="Grocery List" src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `iframe` element has no `title` attribute.
+
+```html
+<iframe src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+#### Failed Example 2
+
+This `iframe` element has an empty (`""`) `title` [attribute value][].
+
+```html
+<iframe title="" src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+#### Failed Example 3
+
+This `iframe` element has `title` [attribute value][] that consists of only [ASCII whitespace][].
+
+```html
+<iframe title="  " src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+#### Failed Example 4
+
+This `iframe` element should use the `title` attribute, instead of using `aria-label`.
+
+```html
+<iframe aria-label="Grocery list" src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+#### Failed Example 5
+
+This `iframe` element with a [semantic role][] of `article` should use the `title` attribute, instead of using `aria-labelledby`.
+
+```html
+<div id="frame-title-helper">Grocery List</div>
+<iframe role="article" aria-labelledby="frame-title-helper" src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+#### Failed Example 6
+
+This `iframe` element should have a `title` attribute, or be removed from [sequential focus navigation][].
+
+```html
+<iframe src="/test-assets/SC4-1-2-frame-doc.html" role="none"> </iframe>
+```
+
+#### Failed Example 7
+
+This `iframe` element should have a `title` attribute, or be removed as a container by setting an [explicit role][] of `presentation` or `none`, or by setting `aria-hidden` to `true`.
+
+```html
+<iframe tabindex="-1" src="/test-assets/SC4-1-2-frame-doc.html"> </iframe>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+This `iframe` is not part of [sequential focus navigation][] and has no [content][] because its CSS `display` property is `none`.
+
+```html
+<iframe style="display:none;" src="/test-assets/SC4-1-2-frame-doc.html"></iframe>
+```
+
+#### Inapplicable Example 2
+
+This `iframe` is not part of [sequential focus navigation][] and has no [content][] because its CSS `visibility` property is `hidden`.
```

Comment:
```suggestion
This `iframe` is not part of [sequential focus navigation][] and has a [hidden state][] of `true` because its CSS `visibility` property is `hidden`.
```